### DISCUSSION TITLE
Residual separation

### DISCUSSION THREAD

**Discussion by edgarcin95** [asteroid-team/asteroid#606] [Q&A] [answered]
Is there a way to get the residual audio from an inference?
I only work with 1 source for music source separation.
I'm getting inferences with CLI (asteroid-infer) but I wish to have, in addition to the separated source, the audio that wasn't separated.

**Comment by mpariente** [comment, marked as answer]
Do you use your own pretrained model? 

You would need the model to return the unseparated audio from the model but a simple class wrapper with a forward like 
```python
# Pseudo code
def forward(self, x): 
    estimate = self.model(x)
    remaining = x - estimate
    return torch.cat([estimate, remaining])
```
would do the trick I guess.

**Comment by edgarcin95** [comment]
I used a DCUNET model and trained with my own dataset.
